src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -544,6 +544,28 @@ private void acknowledgementRoutes(){
                     }
                 });
     }
+    private void validationRoutes()
+    {
+        from("rest:GET:/accounts/{identifierType}")
+                .id("validation-confirm")
+                .log(LoggingLevel.INFO, "Validation Check for identifier type ${header.identifierType}")
+                .process(e -> {
+                            Integer identifierType = e.getIn().getHeader("identifierType", Integer.class);
+                            String finalAmsVal = e.getIn().getHeader("ams", String.class).toLowerCase();
+                            if (finalAmsVal == null || (!finalAmsVal.equalsIgnoreCase("paygops") || !finalAmsVal.equalsIgnoreCase("roster")))
```

Comment:
Null check is very low level, use string util functions. The hard coded check for multiple implementation should be avoided. Are these strings( paygiops and roster ) hard-coded every where in the system? Can't we use them from configurations?
Multiple implementations should be handled in a object oriented way by using interface and implementations

Reply:
Ok I will make changes for null checks and update the PR.